general/array.hpp

```diff
@@ -103,14 +98,17 @@ class Array
    explicit inline Array(std::initializer_list<CT> values);
 
    /// Move constructor ("steals" data from 'src')
-   inline Array(Array<T> &&src) : Array() { Swap(src, *this); }
+   Array(Array<T> &&src) = default;
```

Comment:
Does anyone know if the default implementation resets `src.size` to `0`? The requirement is that `src` is in a valid state after the assignment, however, empty `data` with non-zero `size` is not exactly valid. Invoking the destructor is ok but some other operations can lead to errors.

Reply:
No, I don't think `src.size` will be reset to zero - I think it will be unchanged. The implementation proposed here is probably better: https://github.com/mfem/mfem/pull/4028#discussion_r2264006726

Reply:
Resolved in 5a7edc9548de0d9aaa82be63e92533cb19394534.